File: Counter/src/main/java/com/kchmielewski/java/spring/counter/CounterConfig.java
package com.kchmielewski.java.spring.counter;

import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Scope;

import java.util.concurrent.atomic.AtomicInteger;

@Configuration
@ComponentScan(basePackageClasses = CounterConfig.class)
public class CounterConfig {

    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
    @Bean
    public AtomicInteger atomicInteger() {
        return new AtomicInteger();
    }
}

File: Counter/src/main/kotlin/com/kchmielewski/kotlin/spring/counter/CounterConfig.kt
package com.kchmielewski.kotlin.spring.counter

import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Scope
import java.util.concurrent.atomic.AtomicInteger

/**
 * Configuration for counter scope.
 */
@Configuration
@ComponentScan(basePackageClasses = arrayOf(CounterConfig::class))
open class CounterConfig {

    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
    @Bean
    open fun atomicInteger() = AtomicInteger()
}
